pkg/utils/gateway_wrapper.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	MultiClusterIPAddressType       gatewayv1beta1.AddressType = "kuadrant.io/MultiClusterIPAddress"
+	MultiClusterHostnameAddressType gatewayv1beta1.AddressType = "kuadrant.io/MultiClusterHostnameAddress"
+)
+
+type GatewayWrapper struct {
+	*gatewayv1beta1.Gateway
+	isMultiCluster bool
+}
+
+func NewGatewayWrapper(g *gatewayv1beta1.Gateway) (*GatewayWrapper, error) {
+	gw := &GatewayWrapper{Gateway: g, isMultiCluster: false}
+
+	for i, address := range gw.Status.Addresses {
+		if i == 0 {
+			gw.isMultiCluster = isMultiClusterAddressType(*address.Type)
+			continue
+		}
+		if gw.isMultiCluster == isMultiClusterAddressType(*address.Type) {
+			continue
+		}
+		return nil, fmt.Errorf("gateway is invalid: inconsistent status addresses")
+
+	}
+	return gw, nil
+}
+
+func isMultiClusterAddressType(addressType gatewayv1beta1.AddressType) bool {
+	return addressType == MultiClusterIPAddressType || addressType == MultiClusterHostnameAddressType
+}
+
+func (a *GatewayWrapper) GetKind() string {
```

Comment:
"g" would probably be a better receiver name

Reply:
wondering why it was `a` initially :thinking:  you are correct - now it will be `g`